client_inventory.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+
+	"github.com/mendersoftware/log"
+	"github.com/pkg/errors"
+)
+
+type InventoryReporter interface {
+	Submit(api ApiRequester, server string, data interface{}) error
+}
+
+type InventoryClient struct {
+}
+
+func NewInventoryClient() InventoryReporter {
+	return &InventoryClient{}
+}
+
+// Report status information to the backend
+func (i *InventoryClient) Submit(api ApiRequester, url string, data interface{}) error {
```

Comment:
I do not see how that improves things. Instead of 4-5 client structs, we'll have 4-5 `makeFunc`s. Plus, you need to account for interpretation of return statuses. For instance log & status expect 204, inventory expects 200, update expects 204. You'd have to wrap that, again, in some endpoint specific code that would be passed as (5th?) argument to `Request()`. Not really useful, don't you agree?

Besides, each client should be so simple that we could easily drop instantiaing update/log/status clients in `Mender` constructor, and just create them on demand, even using a struct initializer, like you're supposed to do it using the 'Go-way'.